test/parallel/sdk.API.submitableExtrinsic.test.ts

```diff
@@ -0,0 +1,321 @@
+/*
+ *
+ * @group sdk
+ */
+import { jest } from "@jest/globals";
+import { Keyring } from "@polkadot/api";
+import { getApi, getMangataInstance, initApi } from "../../utils/api";
+import { Assets } from "../../utils/Assets";
+import { BN } from "@polkadot/util";
+import { setupApi, setupUsers } from "../../utils/setup";
+import { Sudo } from "../../utils/sudo";
+import { AssetWallet, User } from "../../utils/User";
+import {
+  getEnvironmentRequiredVars,
+  getUserBalanceOfToken,
+} from "../../utils/utils";
+import { Xyk } from "../../utils/xyk";
+import { MGA_ASSET_ID } from "../../utils/Constants";
+import {
+  activateLiquidity,
+  getBalanceOfPool,
+  getLiquidityAssetId,
+  getRewardsInfo,
+} from "../../utils/tx";
+import {
+  BN_BILLION,
+  BN_HUNDRED,
+  BN_ONE,
+  BN_ZERO,
+  MangataInstance,
+  MangataSubmittableExtrinsic,
+  signTx,
+} from "@mangata-finance/sdk";
+import { waitForRewards } from "../../utils/eventListeners";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(2500000);
+process.env.NODE_ENV = "test";
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+let testUser: User;
+let sudo: User;
+let keyring: Keyring;
+let token1: BN;
+let liqId: BN;
+let mangata: MangataInstance;
+const defaultCurrencyValue = new BN(250000);
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+  keyring = new Keyring({ type: "sr25519" });
+
+  // setup users
+  sudo = new User(keyring, sudoUserName);
+
+  [testUser] = setupUsers();
+
+  await setupApi();
+
+  [token1] = await Assets.setupUserWithCurrencies(
+    sudo,
+    [defaultCurrencyValue],
+    sudo
+  );
+
+  mangata = await getMangataInstance();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.FinalizeTge(),
+    Assets.initIssuance(),
+    Assets.mintNative(testUser),
+    Assets.mintToken(token1, testUser, Assets.DEFAULT_AMOUNT),
+    Sudo.sudoAs(
+      testUser,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        Assets.DEFAULT_AMOUNT.divn(2),
+        token1,
+        Assets.DEFAULT_AMOUNT.divn(2)
+      )
+    )
+  );
+
+  liqId = await getLiquidityAssetId(MGA_ASSET_ID, token1);
+
+  await Sudo.batchAsSudoFinalized(Assets.promotePool(liqId.toNumber(), 20));
+
+  testUser.addAsset(liqId);
+  testUser.addAsset(MGA_ASSET_ID);
+  testUser.addAsset(token1);
+});
+
+test("activate some Liquidity using SDK THEN claim rewards THEN deactivate Liquidity", async () => {
+  await testUser.refreshAmounts(AssetWallet.BEFORE);
+
+  const tx1 = await mangata.submitableExtrinsic.activateLiquidity(
```

Comment:
why awaiting a non-async method?

Reply:
<img width="441" alt="image" src="https://github.com/mangata-finance/mangata-e2e/assets/105198474/286ee9dd-3861-4ef2-8bec-84625d6489f1">
<img width="513" alt="image" src="https://github.com/mangata-finance/mangata-e2e/assets/105198474/143a227a-1b84-40be-870c-1c06f7ba5567">
Are we? I thought if we received different results we need to use a wait function

Reply:
but you are not getting any result on this tx1, only when you sign and submit the tx you need to use await.  On this tx1 you are only creating the extrinsic object. AFIK you do not need to await here, only in the signSubmittableExtrinsic,